examples/sugarscape_g1mt/sugarscape_g1mt/model.py

```diff
@@ -0,0 +1,224 @@
+
+    def __init__(
+        self,
+        width=50,
+        height=50,
+        initial_population=200,
+        endowment_min=25,
+        endowment_max=50,
+        metabolism_min=1,
+        metabolism_max=5,
+        vision_min=1,
+        vision_max=5,
+        seed=96,
+    ):
+        """
+        Create a new Constant Growback model with the given parameters.
+        Args:
+            initial_population: Number of population to start with
+        """
+
+        # Set parameters
+        self.width = width
+        self.height = height
+        self.initial_population = initial_population
+        self.endowment_min = endowment_min
+        self.endowment_max = endowment_max
+        self.metabolism_min = metabolism_min
+        self.metabolism_max = metabolism_max
+        self.vision_min = vision_min
+        self.vision_max = vision_max
+
+        self.schedule = mesa.time.RandomActivationByType(self)
+        self.grid = mesa.space.MultiGrid(
+            self.width, self.height, torus=True
+        )  # page 22 GAS
+        self.datacollector = mesa.DataCollector(
+            model_reporters={
+                "Trader": lambda m: m.schedule.get_type_count(Trader),
+                "Trade volume": lambda m: sum(
+                    len(a.trade_partners)
+                    for a in m.schedule.agents_by_type[Trader].values()
+                ),
+                "Price": lambda m: list(
+                    itertools.chain.from_iterable(
+                        [a.prices for a in m.schedule.agents_by_type[Trader].values()]
+                    )
+                ),
+            },
+            agent_reporters={"trade_partners": lambda a: get_trade(a)},
+        )
+
+        # Create sugar
+        sugar_distribution = np.genfromtxt("sugarscape_g1mt/sugar-map.txt")
+        spice_distribution = np.flip(sugar_distribution, 1)
+        # ensure each agent has a unique _id
+        agent_id = 0
+        for _, x, y in self.grid.coord_iter():
+            max_sugar = sugar_distribution[x, y]
+            if max_sugar > 0:
+                sugar = Sugar(agent_id, self, (x, y), max_sugar)
+                self.grid.place_agent(sugar, (x, y))
+                self.schedule.add(sugar)
+                agent_id += 1
+            # Same, but spice
+            max_spice = spice_distribution[x, y]
+            if max_spice > 0:
+                spice = Spice(agent_id, self, (x, y), max_spice)
+                self.grid.place_agent(spice, (x, y))
+                self.schedule.add(spice)
+                agent_id += 1
+
+        # Create agent:
+        for i in range(self.initial_population):
+            x = self.random.randrange(self.width)
+            y = self.random.randrange(self.height)
+            # See GAS page 108 for parameters initialization.
+            # Each agent is endowed by a random amount of sugar and spice
+            sugar = int(self.random.uniform(self.endowment_min, self.endowment_max + 1))
```

Comment:
Reminder to remove the `+ 1`.

Reply:
I just saw in the video that the `int()` excludes the endpoint because it's always rounded down. Maybe it should be `randrange` instead?

Reply:
It works either way, but with `randrange`, we don't have to explain why we need to add `+ ` even though `random.uniform` includes the endpoint.

Reply:
Unless you object I am ok with living it in because I think it is a good teaching point on the importance of documentation and understanding how functions work. 

The more I do this, the more I viewing it as an advanced Python tutorial as well, that addresses thing I learned over years.

Let me know what you think.

Reply:
I pick `randrange`, because in addition to having to explain the endpoint, you also have to explain why `sugar` and `spice` are initialized differently from `x`, and `y`, even though both are `int` -- based on Occam's razor. The teaching point about documentation can be done in another section of the code. This way, we have shorter video / have extra time for other stuff.